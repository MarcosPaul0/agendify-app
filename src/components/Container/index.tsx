import { StatusBar } from 'expo-status-bar';
import { SafeAreaView } from 'react-native';
import {
  useFonts,
  Roboto_400Regular,
  Roboto_700Bold,
} from '@expo-google-fonts/roboto';
import { SplashScreen } from 'expo-router';
import { IScreenContainerProps } from './interfaces/screenContainerProps.interface';
import { CONTAINER_BG_COLOR } from './constants/containerBgColor.constant';

SplashScreen.preventAutoHideAsync();

export function Container({
  children,
  bgColor = 'gray',
  hasMarginTop = true,
}: IScreenContainerProps) {
  const [fontsLoaded] = useFonts({
    Roboto_400Regular,
    Roboto_700Bold,
  });

  if (!fontsLoaded) {
    return <SplashScreen />;
  }

  const marginTop = hasMarginTop && 'pt-6';

  return (
    <>
      <StatusBar style="dark" />

      <SafeAreaView
        className={`
          flex-1 items-center bg-midnight ${marginTop} ${CONTAINER_BG_COLOR[bgColor]}
        `}
      >
        {children}
      </SafeAreaView>
    </>
  );
}
