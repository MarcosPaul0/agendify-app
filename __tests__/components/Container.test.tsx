import { render } from '@testing-library/react-native';
import { Container } from '@components/Container';
import { Text } from 'react-native';

describe('Container', () => {
  it('Match with snapshot', async () => {
    const tree = render(
      <Container>
        <Text>Fake Children</Text>
      </Container>
    );

    expect(tree).toMatchSnapshot();
  });
});
