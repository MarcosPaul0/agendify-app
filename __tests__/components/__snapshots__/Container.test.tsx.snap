// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Match with snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "alignItems": "center",
      },
      {
        "backgroundColor": "#F8F9FA",
      },
    ]
  }
>
  <Text>
    Fake Children
  </Text>
</RCTSafeAreaView>
`;
